lib/puppet/module/task.rb

```diff
@@ -76,11 +76,23 @@ def self.get_file_details(path, mod)
     # Find task's required lib files and retrieve paths for both 'files' and 'implementation:files' metadata keys
     def self.find_extra_files(metadata, envname = nil)
       return [] if metadata.nil?
+      
+      files = metadata.fetch('files', [])
+      unless files.is_a?(Array)
```

Comment:
Is using `json-schema` validation for this viable? https://github.com/ruby-json-schema/json-schema